sources/packages/backend/apps/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -442,27 +454,38 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
    * @param auditUserId user that should be considered the one that is
    * causing the changes.
    * @param applicationId application id.
-   * @returns a new student restriction object, that need to be saved.
+   * @returns a new student restriction objects, that need to be saved.
    */
   async getPartTimeStudentRestrictions(
     scholasticStandingData: ScholasticStanding,
     studentId: number,
     auditUserId: number,
     applicationId: number,
-  ): Promise<StudentRestriction | undefined> {
+  ): Promise<StudentRestriction[]> {
+    const studentRestriction: StudentRestriction[] = [];
     if (
       [
         StudentScholasticStandingChangeType.StudentDidNotCompleteProgram,
         StudentScholasticStandingChangeType.StudentWithdrewFromProgram,
       ].includes(scholasticStandingData.scholasticStandingChangeType)
     ) {
-      return this.studentRestrictionSharedService.createRestrictionToSave(
-        studentId,
-        RestrictionCode.PTSSR,
-        auditUserId,
-        applicationId,
+      // Create an array to hold the restriction promises.
+      const restrictionPromises: Promise<StudentRestriction>[] = [];
+      PART_TIME_SCHOLASTIC_STANDING_RESTRICTIONS.forEach(
+        (partTimeScholasticStandingRestriction) => {
+          restrictionPromises.push(
+            this.studentRestrictionSharedService.createRestrictionToSave(
+              studentId,
+              partTimeScholasticStandingRestriction,
+              auditUserId,
+              applicationId,
+            ),
+          );
+        },
       );
+      studentRestriction.push(...(await Promise.all(restrictionPromises)));
```

Comment:
`Map` is preferred over a "forEach" + "array declarion" + "push".
Can you please adjust the code as below.
```ts
// Create an array to hold the restriction promises.
const restrictionPromises =
  PART_TIME_SCHOLASTIC_STANDING_RESTRICTIONS.map((restrictionCode) =>
    this.studentRestrictionSharedService.createRestrictionToSave(
      studentId,
      restrictionCode,
      auditUserId,
      applicationId,
    ),
  );
const restrictions = await Promise.all(restrictionPromises);
studentRestriction.push(...restrictions);
```